lib/KangarooTwelve.c

```diff
@@ -40,20 +39,24 @@ static void TurboSHAKE128_Absorb(TurboSHAKE128_Instance *instance, const unsigne
     size_t i, j;
     uint8_t partialBlock;
     const unsigned char *curData;
-    const uint8_t rateInBytes = K12_rateInBytes;
+
+    const uint8_t rateInBytes = instance->rate/8;
 
     assert(instance->squeezing == 0);
 
     i = 0;
     curData = data;
     while(i < dataByteLen) {
         if ((instance->byteIOIndex == 0) && (dataByteLen-i >= rateInBytes)) {
-#ifdef KeccakP1600_12rounds_FastLoop_supported
-            /* processing full blocks first */
-            j = KeccakP1600_12rounds_FastLoop_Absorb(instance->state, K12_rateInLanes, curData, dataByteLen - i);
-            i += j;
-            curData += j;
-#endif
+            // TODO: the below function fails with KT256 because it assumes that the 
+            //  laneCount is 21, however, in the case of KT256, it's 17. So for now,
+            //  I am commenting it till the fix is found.
+// #ifdef KeccakP1600_12rounds_FastLoop_supported
+//             /* processing full blocks first */
+//             j = KeccakP1600_12rounds_FastLoop_Absorb(instance->state, instance->rate/64, curData, dataByteLen - i);
+//             i += j;
+//             curData += j;
```

Comment:
Hello Prof @gvanas, here I had to comment out this part for the tests to match the [Test Vectors](https://datatracker.ietf.org/doc/html/draft-irtf-cfrg-kangarootwelve#section-5) provided in the draft.

After a bit of debugging, and comparing the  _KeccakP1600_AVX512_12rounds_FastLoop_Absorb_ implementation in [K12](https://github.com/XKCP/K12/blob/96738535cbf7495285c767b02ac35d589c251e26/lib/Optimized64/KeccakP-1600-opt64.c#L1003) vs that of [XKCP](https://github.com/XKCP/XKCP/blob/908262a95140e21598ca0a703e45d770a168e8ec/lib/low/KeccakP-1600/plain-64bits/KeccakP-1600-opt64.c#L546), I have tried to remove the `assert(laneCount == 21)` from the K12's implementation since it was the only difference, and since for KT256, `laneCount` will be 17. However, this haven't solved the problem. 

I will do more debugging to try to find the source of the problem, please let me know if I missed something here in the meantime. 

Thank you